modules/javafx.graphics/src/main/java/javafx/css/TransitionEvent.java

```diff
@@ -94,14 +99,17 @@ public final class TransitionEvent extends Event {
      *
      * @param eventType the event type
      * @param property the {@code StyleableProperty} that is targeted by the transition
+     * @param propertyName the name of the targeted CSS property or sub-property
      * @param elapsedTime the time that has elapsed since the transition has entered its active period
      * @throws NullPointerException if {@code eventType}, {@code property} or {@code elapsedTime} is {@code null}
      */
     public TransitionEvent(EventType<? extends Event> eventType,
                            StyleableProperty<?> property,
+                           String propertyName,
                            Duration elapsedTime) {
         super(Objects.requireNonNull(eventType, "eventType cannot be null"));
         this.property = Objects.requireNonNull(property, "property cannot be null");
+        this.propertyName = Objects.requireNonNull(propertyName, "propertyName cannot be null");
```

Comment:
Can this be different from `property.getCssMetadata().getProperty()`?  What does it mean if it is?

Reply:
It can be different. For example, a transition targeting `-fx-border-color` will show `property.getCssMetaData().getProperty()` == `-fx-region-border`.